### PR TITLE
Feature archive

### LINKED ISSUE #11: Delete roles on /unkusss
**Is your feature request related to a problem? Please describe.**
Currently only the permissions are removed. This means that the guild still saves the channels if they are empty.

**Describe the solution you'd like**
Kilian should automatically remove channels if no one has permission to it any more and additionally they are empty. If they are not empty they should get moved to an archive. If someone joins again the channel should be retrieved from the archive again.

### PR DESCRIPTION
**Pull request by xMuffin**
When leaving a channel or using /unkusss, channels will now be moved to an "Archived" category. 
Also unused roles and channels will now be deleted. 
Added ctx.defer, so Kilian is usable with poor connection
Moved some ctx.send, so Kilian first does his job, then says "Welcome to Kilian papa xoxo"

Closes #11

### CODE REVIEW

File: sql_queries.py [hunk 1]
```diff
@@ -42,6 +46,8 @@
 
 delete_student_course = "DELETE FROM student_courses WHERE (discord_id, lva_nr, semester) = (?,?,?)"
 
+delete_archived = "DELETE FROM archived WHERE (guild_id, channel_id) = (?,?,?)"
+
```
Comment (rechen-werk) [inline sql_queries.py:50]: Three ? for two parameters?

File: sql_queries.py [hunk 2]
```diff
@@ -100,12 +106,20 @@
 create_categories = "CREATE TABLE IF NOT EXISTS category(" \
                     "guild_id TEXT NOT NULL," \
                     "category_id TEXT NOT NULL," \
-                    "PRIMARY KEY (guild_id)" \
+                    "archive_id TEXT," \
+                    "PRIMARY KEY (guild_id, category_id)" \
                     ")"
```
Comment (rechen-werk) [inline sql_queries.py:111]: Why is category_id in the primary key now?

Comment (xMuffin) [reply sql_queries.py:111]: fixed

File: kilian.py [hunk 3]
```diff
@@ -110,19 +118,18 @@ async def kusss(ctx: interactions.CommandContext, link: str, studentnumber: str
...
+        await ctx.defer(ephemeral=True)
         user_id = str(ctx.author.id)
         guild_id = str(ctx.guild_id)
-        if not database.is_kusss(str(ctx.author.id)):
-            await ctx.send(f"Seems that your are not registered {ctx.author.name}! You can join the club anytime with "
-                           f"`/kusss`!")
-            return
-        if database.is_kusss(str(ctx.author.id)):
-            await ctx.send("A pity to see you leave " + ctx.author.name + ". You can join the club anytime with "
-                                                                          "`/kusss`!")
```
[the remark this answers is not in the document]
Comment (xMuffin) [reply kilian.py:124]: fixed :))))))))))))))

**Review by rechen-werk — APPROVED** [review]
Thank you for your service :)

**Comment by xMuffin** [comment]
pleas no bad

**Pull request merged**